code/items/weapons/Knife.cs

```diff
@@ -57,10 +57,7 @@ public override void AttackPrimary()
         {
             (Owner as AnimEntity).SetAnimBool("b_attack", true);
 
-            using (Prediction.Off())
-            {
-                ShootEffects();
-            }
+            ShootEffects();
```

Comment:
Good catch, seems like we shouldn't be turning off prediction for this.

https://github.com/Facepunch/sbox-hidden/blob/dbe2903ec3af243715a34275414f63fa49b88400/code/weapons/SMG.cs#L36

Reply:
Maybe this helps in some way. Anyways, doesn't explain the other issues we had with hearing yourself talking, animation issue etc. (this is just for the shooting effects).